index.d.ts

```diff
@@ -2,84 +2,90 @@ import { Redis } from "ioredis";
 }
 
 interface Logger {
-	debug: (input: LoggerInput) => void;
-	warn: (input: LoggerInput) => void;
-	error: (input: LoggerInput) => void;
+  debug: (input: LoggerInput) => void;
+  warn: (input: LoggerInput) => void;
+  error: (input: LoggerInput) => void;
 }
 interface StorageRedisOptions {
-	client: Redis;
-	log?: Logger;
-	invalidation?: { referencesTTL: number } | boolean;
+  client: Redis;
+  log?: Logger;
+  invalidation?: { referencesTTL: number } | boolean;
 }
 
 interface StorageMemoryOptions {
-	size?: number;
-	log?: Logger;
-	invalidation?: boolean;
+  size?: number;
+  log?: Logger;
+  invalidation?: boolean;
 }
 
 interface DataTransformer {
-	serialize: (data: any) => any;
-	deserialize: (data: any) => any;
+  serialize: (data: any) => any;
+  deserialize: (data: any) => any;
 }
 
 type Events = {
-	onDedupe?: (key: string) => void;
-	onError?: (err: any) => void;
-	onHit?: (key: string) => void;
-	onMiss?: (key: string) => void;
+  onDedupe?: (key: string) => void;
+  onError?: (err: any) => void;
+  onHit?: (key: string) => void;
+  onMiss?: (key: string) => void;
 };
 
 type StorageInputRedis = {
-	type: "redis";
-	options?: StorageRedisOptions;
+  type: "redis";
+  options?: StorageRedisOptions;
 };
 
 type StorageInputMemory = {
-	type: "memory";
-	options?: StorageMemoryOptions;
+  type: "memory";
+  options?: StorageMemoryOptions;
 };
 
 declare class StorageInterface {
-	constructor(options: any);
-
-	get(key: string): Promise<undefined | any>;
-	set(key: string, value: any, ttl: number, references?: References): Promise<void>;
-	remove(key: string): Promise<void>;
-	invalidate(references: References): Promise<void>;
-	clear(name: string): Promise<void>;
-	refresh(): Promise<void>;
+  constructor(options: any);
+
+  get(key: string): Promise<undefined | any>;
+  set(key: string, value: any, ttl: number, references?: References): Promise<void>;
+  remove(key: string): Promise<void>;
+  invalidate(references: References): Promise<void>;
+  clear(name: string): Promise<void>;
+  refresh(): Promise<void>;
 }
 
 declare function createCache(
-	options?: {
-		storage?: StorageInputRedis | StorageInputMemory;
-		ttl?: number | ((result: unknown) => number);
-		transformer?: DataTransformer;
-		stale?: number | ((result: unknown) => number);
-	} & Events,
+  options?: {
+    storage?: StorageInputRedis | StorageInputMemory;
+    ttl?: number | ((result: unknown) => number);
+    transformer?: DataTransformer;
+    stale?: number | ((result: unknown) => number);
+  } & Events,
 ): Cache;
 
 declare class Cache {
   constructor(
     options: {
       ttl: number | ((result: unknown) => number);
-      storage: StorageOptionsType;
+      stale?: number | ((result: unknown) => number);
+      storage: StorageInterface;
     } & Events
   );
 
   define<T extends (args: any) => any, N extends string, S extends this>(
     name: N,
     opts: {
-      storage?: StorageOptionsType;
+      storage?: StorageOptions;
```

Comment:
Defind function expects an object with `{ type, options }`, not just string type.